From 7fc0ca63eb07bcd20a580e0be7408a44f4a4065b Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Fri, 9 Oct 2009 11:10:50 +0800
Subject: [PATCH 121/505] ENGR00117122 VPU: Fix interlace information report
 incorrect issue

1. Reverse interlace information description for H264 decoder.
2. Fix MPEG2 interlaced frame judgement.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 test/mxc_vpu_test/dec.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/mxc_vpu_test/dec.c b/test/mxc_vpu_test/dec.c
index 7d645e7..9f75561 100644
--- a/test/mxc_vpu_test/dec.c
+++ b/test/mxc_vpu_test/dec.c
@@ -932,7 +932,7 @@ decoder_start(struct decode *dec)
 						  outinfo.topFieldFirst, decIndex);
 				}
 			} else if ((dec->cmdl->format != STD_MPEG4) && (dec->cmdl->format != STD_RV)){
-				if (!outinfo.interlacedFrame) {
+				if (outinfo.interlacedFrame) {
 					if (outinfo.pictureStructure == 1)
 						field = V4L2_FIELD_TOP;
 					else if (outinfo.pictureStructure == 2)
@@ -1429,7 +1429,7 @@ decoder_parse(struct decode *dec)
 	if (initinfo.streamInfoObtained) {
 		switch (dec->cmdl->format) {
 		case STD_AVC:
-			info_msg("H.264 Profile: %d Level: %d FrameMbsOnlyFlag: %d\n",
+			info_msg("H.264 Profile: %d Level: %d Interlace: %d\n",
 				initinfo.profile, initinfo.level, initinfo.interlace);
 
 			if (initinfo.aspectRateInfo) {
-- 
1.8.0

